util.c

```diff
@@ -691,7 +692,7 @@ int insert_image(struct image *image, struct image *sub,
 		 * have an extent that starts beyond size.
 		 */
 		len = min(len, size);
-		ret = write_zeroes(fd, len, offset);
+		ret = write_bytes(fd, len, offset, byte);
 		if (ret) {
 			image_error(image, "writing %zu zeroes failed: %s\n", len, strerror(-ret));
```

Comment:
just noticed that this error message might need updating

Reply:
fixed.